apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java

```diff
@@ -93,6 +94,99 @@ public void loadSampleA() {
+      assertBeneficiaryMonthly(beneficiaryFromDb, 1990);
+      assertBeneficiaryMonthly(beneficiaryFromDb, 2000);
+      assertBeneficiaryMonthly(beneficiaryFromDb, 2001);
+      assertBeneficiaryMonthly(beneficiaryFromDb, 2003);
+      assertBeneficiaryMonthly(beneficiaryFromDb, 2019);
+      // Set the batch to 10 so all 5 records are in the same batch
+      LoadAppOptions options = CcwRifLoadTestUtils.getLoadOptionsWithBatchSize(10);
+      // Validate HICN and MBI
+      String expectedMbiHash =
+          RifLoader.computeMbiHash(new IdHasher(options.getIdHasherConfig()), "3456789");
+      String expectedHicn =
+          RifLoader.computeHicnHash(new IdHasher(options.getIdHasherConfig()), "543217066U");
+      assertEquals(
+          expectedMbiHash,
+          beneficiaryFromDb.getMbiHash().orElse(null),
+          "Beneficiary has incorrect mbiHash");
+      assertEquals(expectedHicn, beneficiaryFromDb.getHicn(), "Beneficiary has incorrect HICN");
+    } finally {
+      if (entityManager != null) entityManager.close();
+    }
+  }
+
+  /**
+   * Runs {@link RifLoader} against the modified {@link StaticRifResourceGroup#SAMPLE_A} data for an
+   * <code>UPDATE</code> on a {@link Beneficiary} record that has a single file with multiple
+   * reference years for the same bene id, and does not fit in a single batch. The beneMonthly data
+   * should be populated for the bene in the beneMonthly table for each ref year, and HICN and MBIs
+   * are also populated.
+   */
+  @Test
+  public void
+      multipleReferenceYearsInSingleFileForSameBeneInMultipleBatchesExpectBeneMonthlyPopulated() {
+    loadSample(
+        Arrays.asList(StaticRifResourceGroup.SAMPLE_A_MULTIPLE_ENTRIES_SAME_BENE.getResources()));
+
+    validateBeneficiaryAndSkippedCountsInDatabase(1, 0);
+    // Validate bene monthly
+    EntityManagerFactory entityManagerFactory =
+        PipelineTestUtils.get().getPipelineApplicationState().getEntityManagerFactory();
+    EntityManager entityManager = null;
+    try {
+      entityManager = entityManagerFactory.createEntityManager();
+      Beneficiary beneficiaryFromDb = entityManager.find(Beneficiary.class, "567834");
+      // Checks all 12 months are in beneficiary monthlys for that beneficiary, for each 5 records
+      // loaded (5x12)
+      assertEquals(60, beneficiaryFromDb.getBeneficiaryMonthlys().size());
+      // Checks every month in the beneficiary monthly table for each of the years loaded
+      assertBeneficiaryMonthly(beneficiaryFromDb, 1990);
+      assertBeneficiaryMonthly(beneficiaryFromDb, 2000);
+      assertBeneficiaryMonthly(beneficiaryFromDb, 2001);
+      assertBeneficiaryMonthly(beneficiaryFromDb, 2003);
+      assertBeneficiaryMonthly(beneficiaryFromDb, 2019);
+
+      // Set the batch to 10 so all 5 records are in the same batch
+      LoadAppOptions options = CcwRifLoadTestUtils.getLoadOptionsWithBatchSize(10);
```

Comment:
Isn't this test supposed to do more than 1 batch so the batch size should be smaller than 5? Also, how can this influence the batch size here, after the records were already loaded?

Reply:
Good catch! I copy pasted and didn't change the value.

Reply:
Fixed this! Options is now properly passed into loadSample so it gets used, and the batch test's value is set to 2.